src/Controller/Admin/DataObject/DataObjectController.php

```diff
@@ -1310,8 +1311,18 @@ public function saveAction(Request $request): JsonResponse
             if ($request->get('task') !== 'autoSave' && $request->get('task') !== 'unpublish') {
                 $fields = array_keys($object->getClass()->getFieldDefinitions());
                 foreach ($fields as $field) {
-                    $getter  ='get' . ucfirst($field);
-                    if ($object->$getter() !== $objectFromDatabase->$getter()) {
+                    $getter = 'get' . ucfirst($field);
+                    $objectValue = $object->$getter();
+                    $objectFromDatabaseValue = $objectFromDatabase->$getter();
+                    if ($objectValue !== $objectFromDatabaseValue) {
+                        // For Carbon objects, we need to compare the values - not the objects.
+                        if (
+                            $objectValue instanceof Carbon
+                            && $objectFromDatabaseValue instanceof Carbon
```

Comment:
I think we should rather go for the [`EqualComparisonInterface`](https://github.com/pimcore/pimcore/blob/11.x/models/DataObject/ClassDefinition/Data/EqualComparisonInterface.php#L26) of the field definitions, wdyt?

Reply:
Yes, you are completely right.
I adapted it - is that what you had in mind?